python/pyarrow/tests/test_pandas.py

```diff
@@ -3325,6 +3325,21 @@ def test_cast_timestamp_unit():
     assert result.equals(expected)
 
 
+def test_nested_with_timestamp_tz_round_trip():
+    ts = pd.Timestamp.now()
```

Comment:
What timezone does this timestamp have? Is it a naive timestamp? Would be nice explaining it in comments.